### ISSUE TITLE
Uncertain how to use this tool

### ISSUE THREAD

**Issue by dikkedeur** [Sarjuuk/aowow#108]
Its not an issue, since well its me who doesn't understand what you mean.
In the install of the main repo site you mention this.

while maintaining patch order (suffix: 1 -> 9 -> A -> Z)
.. for every locale you are going to use:

<localeCode>/DBFilesClient/
<localeCode>/Interface/WorldMap/
<localeCode>/Interface/FrameXML/GlobalStrings.lua

Now i have compile the MPQextractor. so that is not the problem.
The actual problem is how to use and extract the necessary files.?
Thats rather unclear for me.

I tried a lot of things, but every time it trows an error.

i inmagine i must go to <wowclientdir>Data/enUs/
since that is the locale.. then extract everything maintaining patch order.
So in short, extract in this case.
<wowclientdir>Data/enUS/backup-enUS.MPQ  then
<wowclientdir>Data/enUS/base-enUS.MPQ then
<wowclientdir>Data/enUS/expansion-locale-enUS.MPQ then
<wowclientdir>Data/enUS/expansion-speech-enUS.MPQ  etc.etc.etc.

but no mather what i try.. i dont get any files out of it.. 
Now it can be that i write the commands wrong..

What is the correct way to extract the files?

**Comment by Sarjuuk** [comment]
Here is the README https://github.com/Sarjuuk/MPQExtractor
Just a note. The patching process is not relevant for wotlk era MPQs.

In any case; If you repeatedly get errors, why are you not posting them.

**Comment by dikkedeur** [comment]
Well the errors are probably because i don't understand the command to actually extract the MPQ files.

anyway here we go.

./MPQExtractor -e -o out /home/onno/ftp/files/WOW/Data/enUS/backup-enUS.MPQ 
Invalid argument: -e

or
./MPQExtractor -e "/home/onno/ftp/files/WOW/Data/enUS/" -f -o out backup-enUS.MPQ 
Opening 'backup-enUS.MPQ'...

Extracting files...

Failed to extract the file '/home/onno/ftp/files/WOW/Data/enUS/' in out//home/onno/ftp/files/WOW/Data/enUS/

or

./MPQExtractor -e "/home/onno/ftp/files/WOW/Data/enUS/backup-enUS.MPQ" -f -o out
No MPQ file specified

i just fail to see the logic at this point in how to properly do the commands to extract the needed files.
So thats why i asked what is the proper way to extract or get the files.

thanks for your time..

**Comment by Sarjuuk** [comment]
-e expects a pattern to extract...
`./MPQExtractor -e <path like in aowow readme> -f -c -o <out path> <mpq archive>`

**Comment by dikkedeur** [comment]
tried it as you said.. i hope..

since i know as i used the list command and read the output, the \Interface\GLUES/LOADINGSCREENS should be in 'expansion-locale-enUS.MPQ.

```
Interface\GLUES\Credits\ZulAman6.blp
Interface\GLUES\Credits\ZulAman7.blp
Interface\GLUES\Credits\ZulAman8.blp
Interface\GLUES\LOADINGSCREENS\LOADSCREENCAVERNSTIME.BLP
Interface\GLUES\LOADINGSCREENS\LOADSCREENCOILFANG.BLP
Interface\GLUES\LOADINGSCREENS\LOADSCREENGRUULSLAIR.BLP
Interface\GLUES\LOADINGSCREENS\LOADSCREENSUNWELL.BLP
Interface\GLUES\LOADINGSCREENS\LoadScreenKarazhan.blp
Interface\GLUES\MODELS\UI_MainMenu_BurningCrusade\Green_Grad1.blp
Interface\GLUES\MODELS\UI_MainMenu_BurningCrusade\HellFirePlanet01.blp
```

So did the following:


./MPQExtractor -e \Interface\GLUES/LOADINGSCREENS\ -f -c -o /home/onno/ftp/files/output expansion-locale-enUS.MPQ Opening 'expansion-locale-enUS.MPQ'...

Extracting files...

Failed to extract the file 'InterfaceGLUES/LOADINGSCREENS -f' in /home/onno/ftp/files/output/interfaceglues/loadingscreens -f

**Comment by Sarjuuk** [comment]
i suggest you read the part of the Readme tagged with the giant **IMPORTANT**

**Comment by dikkedeur** [comment]
The loadingscreens was just as example, since i knew that was in that file.

Now to take everything back on track i tried it again.

/home/onno/ftp/files/WOW/Data/enUS# ./MPQExtractor -e "DBFilesClient/" -f -c -o /home/onno/ftp/files/WOW/Data/enUS/out/ patch-enUS.MPQ 

Opening 'patch-enUS.MPQ'...

Extracting files...

Failed to extract the file 'dbfilesclient/' in /home/onno/ftp/files/WOW/Data/enUS/out/dbfilesclient/

for the record the file MPQExtractor must be in the <wowclientdir>/Data/enUS/  folder right??
or must it be in the Data folder?

End must every MPQ be extracted? or just the patch.mpq and patch-2.mpq etc ..

i tried other commands like making a list file with MPQextractor to see whats in the mpq files and that works flawelessly, but extracting is just not working for me.

maybe its my os where its not working correctly, but i doubt it.
for the record: Ubuntu server 16.04

**Comment by Kittnz** [comment]
You can also just extract the files on a windows os. It's pretty easy

**Comment by Sarjuuk** [comment]
again: 
 * paths *within* the archive use backslashes
 * you cant find directories, only files, so you need to use wildcards like *

`-e "dbfilesclient/"` becomes `-e "dbfilesclient\*"` (case doesn't matter)

It is also irrelevant where the executable is located. You just have to match the relative paths accordingly.
And yes, the required paths are scattered through most of the archives, so you will need to check most of them.


just for reference, as i botched this shell script at  some point and never bothered to fix it. 
what to extract from where and in which order
https://gist.github.com/Sarjuuk/e171bf5e7e9f7503a779

**Comment by dikkedeur** [comment]
Thanks Sarjuuk for your help, i managed to extract the file.
thank you so much for your time and patience.
